app/scripts/metamask-controller.js

```diff
@@ -5993,8 +5957,9 @@ export default class MetamaskController extends EventEmitter {
           this.permissionController,
           origin,
         ),
+        // DRY THIS
         requestPermittedChainsPermissionForOrigin: (options) =>
```

Comment:
`requestPermittedChainsPermissionForOrigin` and `requestPermittedChainsPermissionIncrementalForOrigin` should both be the same now and call `requestPermittedChainsPermissionIncremental`. So we can get rid of `requestPermittedChainsPermissionForOrigin` here which will require cleaning up the addEthereumChain handler, switchEthereumChain handler, and the Ethereum-chain-utils `switchChain` function